extensions/cornerstone/src/commandsModule.ts

```diff
@@ -127,6 +128,69 @@ function commandsModule({
   }
 
   const actions = {
+    hydrateSecondaryDisplaySet: async ({ displaySet, viewportId }) => {
+      if (!displaySet) {
+        return;
+      }
+
+      // if segmentation, we can't use the isOverlay, since PMAP is also an overlay but not a segmentation
+      if (['SEG', 'RTSTRUCT'].includes(displaySet.Modality)) {
+        // update the previously stored segmentationPresentation with the new viewportId
+        // presentation so that when we put the referencedDisplaySet back in the viewport
+        // it will have the correct segmentation representation hydrated
+        commandsManager.runCommand('updateStoredSegmentationPresentation', {
+          displaySet,
+          type:
+            displaySet.Modality === 'SEG'
```

Comment:
If you had displaySet.segmentationRepresentation directly then you could just use it here, and the parser already knows about the contour/labelmap distinction so you would simplify this, and would support other new modalities parsing into a segmentationRepresentation.

Reply:
I used the isOverlayDisplaySet property on the displayset, i noticed PMAP does not use it. we can't put segmentation representation in a displaySet since representation is tied to a viewport. One displaySet can be a SEG on two different viewports with different representations (config etc.)